### ISSUE TITLE
Stateful sets stable in 1.9 (v1beta1.StatefulSet)

### ISSUE THREAD

**Issue by owainlewis** [oracle/mysql-operator#106]
We use `v1beta1.StatefulSet` but stateful sets are stable in K8s 1.9. We should probably update our code to reflect this when appropriate. 

https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/

**Comment by prydie** [comment]
Presumably, this would necessarily result in our dropping support for Kubernetes 1.8? That's not really an option in my view until 1.11 is released and even then we'll need to check that there are no internal 1.8 use-cases that we'd be breaking for little/no benefit.

**Comment by owainlewis** [comment]
@prydie yes it would require dropping support and agree it's a *very* minor internal issue. Going to leave this here as a tracking issue if nothing else.

**Comment by lentzi90** [comment]
Hi! I think you should revisit this issue with Kubernetes 1.16 now released. I tried installing the operator in minikube running 1.16 and got the following error: `Error: apiVersion "apps/v1beta1" in mysql-operator/templates/03-deployment.yaml is not available`.
In other words, unless you update the API versions, things will not work on new clusters starting with 1.16.

Changing `apps/v1beta1` to `apps/v1` in the mentioned template solved the installation problem, but it still isn't possible to create a cluster. The following is logged if I try:

```
I1009 13:05:29.854142       1 reflector.go:240] Listing and watching *v1beta1.StatefulSet from github.com/oracle/mysql-operator/vendor/k8s.io/client-go/informers/factory.go:87
E1009 13:05:29.855931       1 reflector.go:205] github.com/oracle/mysql-operator/vendor/k8s.io/client-go/informers/factory.go:87: Failed to list *v1beta1.StatefulSet: the server could not find the requested resource
```

**Comment by owainlewis** [comment]
Thanks @lentzi90. You're correct. We need to look into this and publish a new release.

**Comment by petershaw** [comment]
Any updates so far? Is there a quick workaround for that issue?
